main.js

```diff
@@ -0,0 +1,171 @@
+        id : jsPsych.timelineVariable('id'),
+        item_type : 'FIX_CROSS',
+        uil_save : false
+    }
+};
+
+let present_text = {
+    type : 'spr-moving-window',
+    stimulus : jsPsych.timelineVariable('stimulus'),
+    background_color : "rgb(230, 230, 230)", // light gray
+    font_color : "rgb(0, 0, 0)", // black
+    font_family : "Times New Roman",
+    font_size : 36,
+    width : MIN_WIDTH,
+    height : MIN_HEIGHT,
+    post_trial_gap : ISI,
+    grouping_string : GROUPING_STRING,
+    data : {
+        id : jsPsych.timelineVariable('id'),
+        item_type : jsPsych.timelineVariable('item_type'),
+        uil_save : true
+    }
+}
+
+let question = {
+    type : 'html-button-response',
+    stimulus : jsPsych.timelineVariable('question'),
+    choices : G_QUESTION_CHOICES,
+    data : {
+        id : jsPsych.timelineVariable('id'),
+        item_type : "Q" + jsPsych.timelineVariable('item_type'),
+        expected_answer : jsPsych.timelineVariable('qanswer'),
+        uil_save : true
+    },
+    on_finish: function (data) {
+        let choice = G_QUESTION_CHOICES[data.button_pressed];
+        data.answer = choice;
+        data.correct = choice === data.expected_answer;
+        data.integer_correct = data.correct ? 1 : 0;
+        data.rt = Math.round(data.rt);
+    }
+};
+
+let maybe_question = {
+    timeline: [ question ],
+    conditional_function: function() {
+        let q = jsPsych.timelineVariable('question')();
+        return typeof(q) !== 'undefined' && q.length > 0;
+    }
+};
+
+let end_practice_screen = {
+    type : 'html-keyboard-response',
+    stimulus : PRE_TEST_INSTRUCTION,
+    choices : [KEY_CODE_SPACE],
+    response_ends_trial : true,
+    on_finish: function (data) {
+        data.rt = Math.round(data.rt);
+    }
+};
+
+let end_experiment = {
+    type : 'html-keyboard-response',
+    stimulus : POST_TEST_INSTRUCTION,
+    choices : [],
+    trial_duration : FINISH_TEXT_DUR,
+    on_finish: function (data) {
+        data.rt = Math.round(data.rt);
+    }
+}
+
+function getTimeline(stimuli) {
+    //////////////// timeline /////////////////////////////////
+    let timeline = [];
+
+    // Welcome the participant and guide them through the
+    // consent forms and survey.
+    timeline.push(welcome_screen);
+
+    // Obtain informed consent.
+    timeline.push(consent_procedure);
+
+    // add survey
+    timeline.push(survey_procedure);
+
+    // Add the different parts of the experiment to the timeline
+    timeline.push(instruction_screen_practice);
+
+    let practice = {
+        timeline: [
+            fixcross,
+            present_text,
+            maybe_question
+        ],
+        timeline_variables: PRACTICE_ITEMS
+    };
+
+    timeline.push(practice);
+    timeline.push(end_practice_screen);
+
+    let test = {
+        timeline: [
+            fixcross,
+            present_text,
+            maybe_question
+        ],
+        timeline_variables: stimuli.table
+    }
```

Comment:
I changed it mainly so that we could have all the timeline together without extra layers of function calls. Otherwise I don't think it makes much of a difference.